lib/components/AlbumScreen/preset_chip.dart

```diff
@@ -0,0 +1,153 @@
+import 'dart:math';
+import 'package:flutter/material.dart';
+import 'package:finamp/services/queue_service.dart';
+import 'package:get_it/get_it.dart';
+
+const _radius = Radius.circular(4);
+const _borderRadius = BorderRadius.all(_radius);
+const _height = 36.0;
+final _defaultBackgroundColour = Colors.white.withOpacity(0.1);
+final _queueService = GetIt.instance<QueueService>();
+
+class PresetChips extends StatefulWidget {
+  const PresetChips({
+    Key? key,
+    required this.type,
+    required this.values,
+    required this.activeValue,
+    this.onTap,
+    this.mainColour,
+    this.onPressed,
+  }) : super(key: key);
+
+  // for future preset types other than "speed"
+  final String type;
+
+  final List<double> values;
+  final double activeValue;
+  final Function()? onTap;
+  final Color? mainColour; // used for different background colours
+  final Function()? onPressed;
```

Comment:
`onPresetSelected`, and pass the value of the selected preset :)

Reply:
I see that I can definitely optimize it, but we need it to rebuild the `SpeedMenu` and update the text field, right?

Reply:
We can do it like this, yes.  
Alternatively, you could use a `ValueListenableBuilder` and listen to `FinampSettingsHelper.finampSettingsListener`. This way, every time the speed is changed, the speed menu will rebuild.